sonic-psud/scripts/psud

```diff
@@ -85,6 +86,108 @@ def psu_db_update(psu_tbl, psu_num):
                                            'true' if _wrapper_get_psus_status(psu_index) else 'false')])
         psu_tbl.set(PSU_INFO_KEY_TEMPLATE.format(psu_index), fvs)
 
+
+# try get information from platform API and return a default value if caught NotImplementedError
+def try_get(callback, default=None):
+    """
+    Handy function to invoke the callback and catch NotImplementedError
+    :param callback: Callback to be invoked
+    :param default: Default return value if exception occur
+    :return: Default return value if exception occur else return value of the callback
+    """
+    try:
+        ret = callback()
+        if ret is None:
+            ret = default
+    except NotImplementedError:
+        ret = default
+
+    return ret
+
+def log_on_status_changed(normal_status, normal_log, abnormal_log):
+    """
+    Log when any status changed
+    :param normal_status: Expected status.
+    :param normal_log: Log string for expected status.
+    :param abnormal_log: Log string for unexpected status
+    :return:
+    """
+    if normal_status:
+        logger.log_notice(normal_log)
+    else:
+        logger.log_warning(abnormal_log)
+
+#
+# PSU status ===================================================================
+#
+
+class PsuStatus(object):
+    update_led_color = True
+
+    def __init__(self, psu):
+        self.psu = psu
+        self.presence = True
+        self.power_good = True
+        self.voltage_good = True
+        self.temperature_good = True
+
+    def set_presence(self, presence):
+        """
+        Set and cache PSU presence status
+        :param presence: PSU presence status
+        :return: True if status changed else False
```

Comment:
status not change can be :1. status already as expected no need to change. 2. failed to change.    how do you distinguish these two cases?   same comments to below set functions.

Reply:
For both cases you mentioned, we need not trigger logs and change led color. So I think there is no difference and would be safe to just return False.